assets/src/edit-story/components/panels/document/status/status.js

```diff
@@ -26,60 +26,74 @@ import { __ } from '@web-stories-wp/i18n';
  */
 import { useStory } from '../../../../app/story';
 import { useConfig } from '../../../../app/config';
-import { Row, TextInput, HelperText, RadioGroup } from '../../../form';
+import { Row, RadioGroup } from '../../../form';
 import { SimplePanel } from '../../panel';
+import { Input, Text, THEME_CONSTANTS } from '../../../../../design-system';
 
-const BoxedTextInput = styled(TextInput)`
-  padding: 6px 6px;
-  border-radius: 4px;
-  flex-grow: 1;
-  &:focus {
-    background-color: ${({ theme }) => theme.DEPRECATED_THEME.colors.fg.white};
-  }
+const InputRow = styled(Row)`
+  margin-left: 34px;
+`;
+
+const HelperText = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.secondary};
 `;
 
 function StatusPanel() {
-  const { status = '', password, updateStory } = useStory(
+  const { status = '', password: savedPassword, updateStory } = useStory(
     ({
       state: {
         story: { status, password },
       },
       actions: { updateStory },
     }) => ({ status, password, updateStory })
   );
+  const [password, setPassword] = useState(savedPassword);
+  const passwordError = password && password.length > 20;
+
+  useEffect(() => {
+    // updated displayed password when stored password is updated
+    setPassword(savedPassword);
+  }, [savedPassword]);
 
   const { capabilities } = useConfig();
 
   const visibilityOptions = [
     {
       value: 'draft',
-      name: __('Draft', 'web-stories'),
+      label: __('Draft', 'web-stories'),
       helper: __('Visible to just me', 'web-stories'),
     },
   ];
 
   if (capabilities?.hasPublishAction) {
     visibilityOptions.push({
       value: 'publish',
-      name: __('Public', 'web-stories'),
+      label: __('Public', 'web-stories'),
       helper: __('Visible to everyone', 'web-stories'),
     });
     visibilityOptions.push({
       value: 'private',
-      name: __('Private', 'web-stories'),
+      label: __('Private', 'web-stories'),
       helper: __('Visible to site admins & editors only', 'web-stories'),
     });
   }
 
   const passwordProtected = 'protected';
   // @todo Add this back once we have FE implementation, too.
   /*visibilityOptions.push({
-    name: __('Password Protected', 'web-stories'),
+    label: __('Password Protected', 'web-stories'),
     value: passwordProtected,
+    helper: __('Need password to view', 'web-stories'),
   });*/
 
+  const handleChangePassword = useCallback((evt) => {
+    setPassword(evt.target.value.trim());
+  }, []);
+
   // @todo this should still allow showing the moment where the warning is red, currently just doesn't allow adding more.
-  const handleChangePassword = useCallback(
+  const handleUpdatePassword = useCallback(
```

Comment:
Password is updated on blur now. There's intermediate state tracked in this component.